### PR TITLE
breaking up the dockerfile into smaller steps

### PR DESCRIPTION
**Pull request by michaelraczycki**
**What is this PR about?**
Addresses timeout issue mentioned in #6590 

**Checklist**
Breaks down Dockerfile into smaller steps to possibly prevent the timeout

----
:books: Documentation preview :books:: https://pymc--6713.org.readthedocs.build/en/6713/

### CODE REVIEW

File: scripts/Dockerfile [hunk 1]
```diff
@@ -1,23 +1,28 @@
-FROM jupyter/base-notebook:python-3.9.12
-
-LABEL name="pymc"
-LABEL description="Environment for PyMC version 4"
-
+# Stage 1: Base image
+FROM jupyter/base-notebook:python-3.9.12 as base
+LABEL name="pymc" description="Environment for PyMC version 4"
```
Comment (twiecki) [inline scripts/Dockerfile]:
```suggestion
LABEL name="pymc" description="Environment for PyMC"
```

Comment (michaelraczycki) [reply scripts/Dockerfile]: fixed

**Comment by michaelraczycki** [comment]
Just to clarify, I'm not 100% sure if this will solve problem right away, but since there was no concrete error that would crash the process, but it was just a timeout it seemed like a good first step

**Comment by twiecki** [comment]
Did you try this out?

**Comment by michaelraczycki** [comment]
I've pinpointed the cause of the failed build. The build fails because pytensor is not accessible via conda in docker env, trying to solve it further. @bwengals did you have such an issue in the past?

**Comment by twiecki** [comment]
@michaelraczycki There are unrelated pytensor builds, let's pause this for now until those are resolved.

File: scripts/Dockerfile [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+FROM jupyter/base-notebook:python-3.9.12 as base
+LABEL name="pymc" description="Environment for PyMC"
 ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
-
-# Switch to jovyan to avoid container runs as root
 USER $NB_UID
 
-COPY /conda-envs/environment-dev.yml .
+# Stage 2: Install Conda dependencies
+FROM base as conda
```
Comment (michaelosthege) [inline scripts/Dockerfile:8]: What does this line do?
Does this have something to do with the `as base` part from line 2 ?

There seeems to be a lot of back and forth with these `FROM a as b` lines..

**Comment by ricardoV94** [comment]
@michaelosthege @maresb do you want to revive the PR?

**Comment by maresb** [comment]
Ya, I'd be happy to overhaul this. What's it being used for?

**Comment by ricardoV94** [comment]
> Ya, I'd be happy to overhaul this. What's it being used for?

No idea :)

**Comment by ricardoV94** [comment]
Closing this as stale

**Pull request closed without merging**